src/game/common/rts/team.cpp

```diff
@@ -1492,3 +1496,721 @@ bool Team::Damage_Team_Members(float amount)
+                continue;
+            }
+        } else if (!Loco_Set_Matches(1, surfaces)) {
+            it.Advance();
+            continue;
+        }
+
+        if (!it.Cur()->Is_Effectively_Dead()) {
+            if (!it.Cur()->Is_KindOf(KINDOF_INERT)) {
+                if (it.Cur()->Is_Inside(poly)) {
+                    b2 = true;
+                } else {
+                    b3 = true;
+                }
+
+                b1 = true;
+            }
+        }
+
+        it.Advance();
+    }
+
+    return b1 && b2 && b3;
+}
+
+const Coord3D *Team::Get_Estimate_Team_Position() const
+{
+    DLINK_ITERATOR<Object> it = Iterate_Team_Member_List();
+
+    if (it.Cur() == nullptr) {
+        return nullptr;
+    }
+
+    const Coord3D *pos = it.Cur()->Get_Position();
+
+    if (pos == nullptr) {
+        return nullptr;
+    }
+
+    return pos;
+}
+
+void Team::Delete_Team(bool ignore_dead)
+{
+    if (this == this->Get_Controlling_Player()->Get_Default_Team()) {
+        std::list<Object *> list;
+        DLINK_ITERATOR<Object> it = Iterate_Team_Member_List();
+
+        while (!it.Done()) {
+            Object *obj = it.Cur();
+
+            if (obj != nullptr && obj->Get_Contain() != nullptr) {
+                if (obj->Get_Contain()->Get_Contain_Count() != 0) {
+                    list.push_back(obj);
+                }
+            }
+
+            it.Advance();
+        }
+
+        for (auto it = list.begin(); it != list.end(); it++) {
+            if ((*it)->Get_Contain() != nullptr) {
+                (*it)->Get_Contain()->Remove_All_Contained(false);
+            }
+        }
+    }
+
+    DLINK_ITERATOR<Object> it = Iterate_Team_Member_List();
+
+    while (!it.Done()) {
+        Object *obj = it.Cur();
+
+        if (obj != nullptr && (!ignore_dead || !obj->Is_Effectively_Dead())) {
+            g_theGameLogic->Destroy_Object(obj);
+        }
+
+        it.Advance();
+    }
+}
+
+void Team::Transfer_Units_To(Team *team)
+{
+    if (this != team && team != nullptr) {
+        for (;;) {
+            Object *obj = Get_First_Item_In_Team_Member_List();
+
+            if (obj == nullptr) {
+                break;
+            }
+
+            obj->Set_Team(team);
+        }
+    }
+}
+
+bool Is_In_Build_Variations(const ThingTemplate *tmplate, const ThingTemplate *tmplate2)
+{
+    const std::vector<Utf8String> &variations = tmplate->Get_Build_Variations();
+
+    if (variations.empty()) {
+        return false;
+    }
+
+    for (auto it = variations.begin(); it != variations.end(); it++) {
+        if (tmplate2->Get_Name() == (*it)) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+Object *Team::Try_To_Recruit(const ThingTemplate *tmplate, const Coord3D *pos, float max_dist)
+{
+    Player *player = Get_Controlling_Player();
+    float dist_sqr = max_dist * max_dist;
+    Object *recruit = nullptr;
+
+    for (Object *o = g_theGameLogic->Get_First_Object(); o != nullptr; o = o->Get_Next_Object()) {
```

Comment:
`obj`, like in other functions of this class.

Reply:
fixed.